### PR TITLE
Use test-lib.sh for common test functions

### PR DESCRIPTION
**Pull request by hhorak**
**Warning: this requires https://github.com/sclorg/container-common-scripts/pull/238 to be merged first.**

This also solves the false positive problem that has been in the test script:
The  made the test stopped at the end of run_change_password_test
test, when the code actually expected the old password to be working
(despite the comment said otherwise).

It effectively skipped the last test (documentation check) and what is worse,
the test was still reported as passed, despite the test run_change_password_test
failed (false positive).

The changes in the run_change_password_test test, avoiding usage of errexit,
and usage of the common functions should avoid happening this again.

### CODE REVIEW

**Comment by hhorak** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by hhorak** [comment]
[test]

**Comment by hhorak** [comment]
[test]

**Comment by hhorak** [comment]
[test]

**Comment by hhorak** [comment]
[test]

**Review by phracek — CHANGES_REQUESTED** [review]
I did not hit any issues. Please rebase it and run tests.

**Comment by phracek** [comment]
[test-all]

**Comment by zmiklank** [comment]
@hhorak 
I did not manage to notice this PR, sorry. A few of these changes are therefore already merged to master. This PR would need to be reworked in order to not be in conflict with master branch.

**Comment by phracek** [comment]
I guess, this was already fixed by pull requests: https://github.com/sclorg/redis-container/pull/138/files and https://github.com/sclorg/redis-container/pull/131/files.

**Comment by zmiklank** [comment]
@phracek Only partly. Few of changes in this PR are unique, and weren't merged in PRs you mentioned.

**Comment by hhorak** [comment]
[test]

**Comment by hhorak** [comment]
@phracek @zmiklank I rebased this to include only the missing changes, hope it makes sense now.

File: test/run [hunk 1]
```diff
@@ -28,23 +28,23 @@ run_doc_test
...
-function get_cid() {
-  local id="$1" ; shift || return 1
-  echo $(cat "$CID_FILE_DIR/$id")
-}
-
-function get_container_ip() {
-  local id="$1" ; shift
-  docker inspect --format='{{.NetworkSettings.IPAddress}}' $(get_cid "$id")
-}
+	check_result() {
```
Comment (zmiklank) [inline test/run:31]: Can be removed in favor of: https://github.com/sclorg/container-common-scripts/blob/646af729971641b5f6ec9de977ec13a563b75bbc/test-lib.sh#L1299

File: test/run [hunk 2]
```diff
@@ -28,23 +28,23 @@ run_doc_test
...
+	  local result="$1"
+	  if [[ "$result" != "0" ]]; then
+	    TESTCASE_RESULT=1
+	  fi
+	  return $result
+	}
+
+	function get_cid() {
```
Comment (zmiklank) [inline test/run:39]: Can be removed in favor of:https://github.com/sclorg/container-common-scripts/blob/646af729971641b5f6ec9de977ec13a563b75bbc/test-lib.sh#L335

File: test/run [hunk 3]
```diff
@@ -28,23 +28,23 @@ run_doc_test
...
+	  local id="$1" ; shift || return 1
+	  echo $(cat "$CID_FILE_DIR/$id")
+	}
+
+	function get_container_ip() {
```
Comment (zmiklank) [inline test/run:44]: Can be removed in favor of: https://github.com/sclorg/container-common-scripts/blob/646af729971641b5f6ec9de977ec13a563b75bbc/test-lib.sh#L344

File: test/run [hunk 4]
```diff
@@ -104,8 +104,8 @@ function create_container() {
   cidfile="$CID_FILE_DIR/$name"
   # create container with a cidfile in a directory for cleanup
   local container_id
-  [ "${DEBUG:-0}" -eq 1 ] && echo "DEBUG: docker run ${DOCKER_ARGS:-} --cidfile $cidfile -d \"$@\" $IMAGE_NAME ${CONTAINER_ARGS:-}" >&2
-  container_id="$(docker run ${DOCKER_ARGS:-} --cidfile $cidfile -d "$@" $IMAGE_NAME ${CONTAINER_ARGS:-})"
+  [ "${DEBUG:-0}" -eq 1 ] && echo "DEBUG: docker run ${DOCKER_ARGS:-} --cidfile \"$cidfile\" -d \"$@\" $IMAGE_NAME ${CONTAINER_ARGS:-}" >&2
```
Comment (zmiklank) [inline test/run:107]: This could be in different commit - does not seem related to test-lib functions.

File: test/run [hunk 5]
```diff
@@ -117,21 +117,21 @@ function run_change_password_test() {
...
+  ct_check_testcase_result $?
   test_connection testpass2 bar
-  check_result $?
+  ct_check_testcase_result $?
   # The old password should not work anymore
-  container_ip="$(get_container_ip testpass2)"
-  connection_works "$container_ip" bar
-  check_result $?
+  container_ip="$(ct_get_cip testpass2)"
+  ! connection_works "$container_ip" foo
```
Comment (zmiklank) [inline test/run:133]: This line also seem not connected with test-lib functions usage. Could it be placed in different commit?

**Review by zmiklank — CHANGES_REQUESTED** [review]
Thank you for the PR.

Please, see my comments.

Also, could you please update the submodule in separate commit?

File: test/run (hunk 2 above)
Comment (hhorak) [reply test/run:39]: yep, this was supposed to be removed, will do it shortly.

File: test/run (hunk 1 above)
Comment (hhorak) [reply test/run:31]: yep, this was supposed to be removed, will do it shortly.

File: test/run (hunk 5 above)
Comment (hhorak) [reply test/run:115]: yep, will do

File: test/run (hunk 4 above)
Comment (hhorak) [reply test/run:89]: yep, will do

**Comment by hhorak** [comment]
[test]

**Comment by hhorak** [comment]
[test]

**Comment by hhorak** [comment]
@zmiklank all the comments should be hopefully addressed, thanks for the review.

**Review by zmiklank — APPROVED** [review]
Thank you. LGTM.
